### PR TITLE
[Reviewer: Ellie] Log net-snmp messages to the Clearwater log

### PR DESCRIPTION
**Pull request by rkd-msw**
Hopefully reasonably self-explanatory - netsnmp lets you provide a callback for logging, so I've done that and written a callback that logs to our normal log. (Previously, it went to syslog.)

### CODE REVIEW

File: src/snmp_agent.cpp [hunk 1]
```diff
@@ -49,12 +49,57 @@ void* snmp_thread(void* data)
   return NULL;
 };
 
+int logging_callback(int majorID, int minorID, void *serverarg, void *clientarg)
+{
+  snmp_log_message *log_message = (snmp_log_message *) serverarg;
```
Comment (eleanor-merry) [inline src/snmp_agent.cpp]: spacing looks funny here

File: src/snmp_agent.cpp [hunk 2]
```diff
@@ -49,12 +49,57 @@ void* snmp_thread(void* data)
...
+  int snmp_prio = log_message->priority;
```
Comment (eleanor-merry) [inline src/snmp_agent.cpp]: Can you not just use priority as the name here?

**Comment by eleanor-merry** [comment]
Looks good

File: src/snmp_agent.cpp (hunk 2 above)
Comment (rkd-msw) [inline src/snmp_agent.cpp]: I'd like to be clear that this is switching between the netsnmp priority levels and the Clearwater ones - hence the naming choices.

Comment (rkd-msw) [inline src/snmp_agent.cpp]: Oh, you mean calling them snmp_priority and clearwater_priority. Yes.

**Pull request merged**